File: backend/Models/Entities/AdressBook.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models.Entities;
public class AdressBook
{
    public int Id { get; set; }
    
    [Column(TypeName = "TEXT")]
    public string? YtsUrls { get; set; }  
    
    [Column(TypeName = "TEXT")]
    public string? RarbgUrls { get; set; }
    
    [Column(TypeName = "TEXT")]
    public string? PirateBayUrls { get; set; }
    
    public string? QbitUrl { get; set; }
    
    [Column(TypeName = "TEXT")]
    public string? SeriesDirectories { get; set; }
}

File: frontend/src/types/AdressBook.ts
export interface UrlEntry {
  url: string;
  active: boolean;
}

export interface AdressBook {
  id: number;
  ytsUrls?: UrlEntry[];
  rarbgUrls?: UrlEntry[];
  pirateBayUrls?: UrlEntry[];
  qbitUrl?: string;
  seriesDirectories?: string[];
}

// Helper to parse backend response
export const parseAdressBook = (data: any): AdressBook => ({
  ...data,
  ytsUrls: data.ytsUrls ? JSON.parse(data.ytsUrls) : [],
  rarbgUrls: data.rarbgUrls ? JSON.parse(data.rarbgUrls) : [],
  pirateBayUrls: data.pirateBayUrls ? JSON.parse(data.pirateBayUrls) : [],
  seriesDirectories: data.seriesDirectories
    ? JSON.parse(data.seriesDirectories)
    : [],
});

// Helper to convert to backend format
export const stringifyAdressBook = (data: AdressBook): any => ({
  ...data,
  ytsUrls: data.ytsUrls ? JSON.stringify(data.ytsUrls) : null,
  rarbgUrls: data.rarbgUrls ? JSON.stringify(data.rarbgUrls) : null,
  pirateBayUrls: data.pirateBayUrls ? JSON.stringify(data.pirateBayUrls) : null,
  seriesDirectories: data.seriesDirectories
    ? JSON.stringify(data.seriesDirectories)
    : null,
});
